ui/app/AppLayouts/Chat/components/MessageContextMenu.qml

```diff
@@ -136,22 +141,36 @@ PopupMenu {
 
     Action {
         id: pinAction
-        //% "Unpin"
-        text: pinnedMessage ? qsTrId("unpin") :
-                              //% "Pin"
-                              qsTrId("pin")
+        text: {
+            //
+            if (pinnedMessage) {
+                //% "Unpin"
+                return qsTrId("unpin")
+            }
+            if (!canPin) {
+                return qsTr("You can only pin %1 messages").arg(Constants.maxNumberOfPins)
+            }
+            //% "Pin"
+            return qsTrId("pin")
+
+        }
         onTriggered: {
             if (pinnedMessage) {
                 chatsModel.messageView.unPinMessage(messageId, chatsModel.channelView.activeChannel.id)
                 return
             }
 
+            if (!canPin) {
```

Comment:
Could we just not display the possibility to pin? or  "disable" the action?

Reply:
I don't think we can disable it, because `enabled: false` removes the action.
Also, I don't like not displaying it, because it can become confusing as to why the button disappeared. I'll change it if there are multiple complaints though.